modules/@apostrophecms/schema/ui/apos/components/AposInputRelationship.vue

```diff
@@ -157,29 +157,39 @@ export default {
       this.next = items;
     },
     async input () {
-      if (!this.searching) {
-        if (this.searchTerm.length) {
-          this.searching = true;
-          const list = await apos.http.get(
-            `${apos.modules[this.field.withType].action}?autocomplete=${this.searchTerm}`,
-            {
-              busy: false,
-              draft: true
-            });
-          // filter items already selected
-          this.searchList = list.results.filter(item => {
-            return !this.next.map(i => i._id).includes(item._id);
-          }).map(item => {
-            return {
-              ...item,
-              disabled: this.disableUnpublished && !item.lastPublishedAt
-            };
-          });
-          this.searching = false;
-        } else {
-          this.searchList = [];
-        }
+      if (this.searching) {
+        return;
       }
+
+      if (!this.searchTerm.length) {
+        this.searchList = [];
+        return;
+      }
+
+      const qs = {};
+
+      if (this.field.withType === '@apostrophecms/image') {
+        apos.bus.$emit('piece-relationship-query', qs);
+      }
+
+      this.searching = true;
+      const list = await apos.http.get(
+        `${apos.modules[this.field.withType].action}?autocomplete=${this.searchTerm}`,
```

Comment:
Or rather, please add `autocomplete` to your `qs` object, because this is a good time to fix the bug that autocomplete could contain special characters but isn't being properly escaped.